app/lib/widgets/avatar.dart

```diff
@@ -0,0 +1,188 @@
+  final VoidCallback? onPressed;
+  final UuidValue? gradientKey;
+
+  @override
+  Widget build(BuildContext context) {
+    final foregroundImage = image != null
+        ? CachedMemoryImage.fromImageData(image!)
+        : null;
+    final colors = CustomColorScheme.of(context);
+    final gradient = _AvatarGradient.fromUuid(gradientKey);
+
+    return GestureDetector(
+      onTap: onPressed,
+      child: MouseRegion(
+        cursor: onPressed != null
+            ? SystemMouseCursors.click
+            : MouseCursor.defer,
+        child: SizedBox(
+          width: size,
+          height: size,
+          child: DecoratedBox(
+            decoration: BoxDecoration(
+              shape: BoxShape.circle,
+              gradient: foregroundImage == null
+                  ? LinearGradient(
+                      colors: gradient.colors,
+                      begin: Alignment.topLeft,
+                      end: Alignment.bottomRight,
+                    )
+                  : null,
+              color: foregroundImage != null ? colors.text.quaternary : null,
+            ),
+            child: CircleAvatar(
+              radius: size / 2,
+              backgroundColor: Colors.transparent,
+              foregroundImage: foregroundImage,
+              child: Text(
+                displayName.characters.firstOrNull?.toUpperCase() ?? "",
+                style: TextTheme.of(context).labelMedium!.copyWith(
+                  color: colors.function.white,
+                  fontSize: LabelFontSize.small2.size * size / 28,
+                  fontWeight: FontWeight.w300,
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _AvatarGradient {
+  const _AvatarGradient({required this.start, required this.end});
+
+  final Color start;
+  final Color end;
+
+  List<Color> get colors => [start, end];
+
+  factory _AvatarGradient.fromUuid(UuidValue? uuid) {
+    final index = _gradientIndexForUuid(uuid);
+    final (start, end) = _gradients[index];
+    return _AvatarGradient(start: start, end: end);
+  }
+
+  static const _start = 300;
+  static const _end = 700;
+
+  static final _gradients = [
+    (AppColors.red[_start]!, AppColors.red[_end]!),
+    (AppColors.orange[_start]!, AppColors.orange[_end]!),
+    (AppColors.yellow[_start]!, AppColors.yellow[_end]!),
+    (AppColors.green[_start]!, AppColors.green[_end]!),
+    (AppColors.cyan[_start]!, AppColors.cyan[_end]!),
+    (AppColors.blue[_start]!, AppColors.blue[_end]!),
+    (AppColors.purple[_start]!, AppColors.purple[_end]!),
+    (AppColors.magenta[_start]!, AppColors.magenta[_end]!),
+  ];
+
+  static int _gradientIndexForUuid(UuidValue? uuid) {
+    if (uuid == null) {
+      return 0;
+    }
+    // Cheap uniformity inspired by Java's String.hashCode()
+    var hash = 0;
+    for (final codeUnit in uuid.uuid.codeUnits) {
+      hash = ((hash << 5) + hash) + codeUnit;
+      hash &= 0xFFFFFFFF;
+    }
```

Comment:
Looks like you can just use `uuid.uuid.hashCode`. The implementation https://github.com/dart-lang/sdk/blob/main/runtime/vm/compiler/assembler/assembler_arm.cc#L3514 is similar to yours.